### PR TITLE
Reset the leave status if the employee's vacation is over

### LINKED ISSUE #23: Reset the leave status if the employee's vacation is over

### PR DESCRIPTION
**Pull request by BehindLoader**
close #23

### CODE REVIEW

File: src/viva.js [hunk 1]
```diff
@@ -138,7 +149,34 @@ module.exports = async (robot) => {
...
+          delete state.leaveEnd
+          delete state.requestStatus
+
+          robot.adapter.sendDirect({ user: { name: user.name } }, 'С возвращением из отпуска')
+        }
+      }
+    }
+  }
+
+  robot.respond(/хочу в отпуск$/i, function (msg) {
```
Comment (eugulixes) [inline src/viva.js]: You forgot `\s*`.

File: src/viva.js [hunk 2]
```diff
@@ -138,6 +149,33 @@ module.exports = async (robot) => {
     }
   }
 
+  /**
+   * Check if the employee's vacation is over
+   * @param {Robot} robot Hubot instance
+   * @returns {Void}
+   */
+  function vacationIsOver (robot) {
```
Comment (eugulixes) [inline src/viva.js]: We need a verb here. Name the function `resetLeaveStatus`.

File: src/viva.js [hunk 3]
```diff
@@ -138,6 +149,33 @@ module.exports = async (robot) => {
     }
   }
 
+  /**
+   * Check if the employee's vacation is over
```
Comment (eugulixes) [inline src/viva.js]: Reset the leave status of the users if their vacation is over.

File: src/viva.js [hunk 4]
```diff
@@ -324,7 +362,40 @@ module.exports = async (robot) => {
     }
   })
 
+  /**
+   * Overwrites user's vacation dates
+   * @example viva reset @username DD.MM(beginning) DD.MM(end)
+   */
```
Comment (polina-popova) [inline src/viva.js]: May be we even don't need the beginning date, we need to reset only end date of the test user's vacation without any usual checks.

**Comment by polina-popova** [comment]
Don't forget to update readme also.

File: src/viva.js [hunk 5]
```diff
@@ -324,7 +362,40 @@ module.exports = async (robot) => {
...
+  robot.respond(/(viva reset @?(.+) (\d{1,2}\.\d{1,2}) (\d{1,2}\.\d{1,2}))\s*/i, async (msg) => {
```
Comment (eugulixes) [inline src/viva.js]: Limit the use of the command to admin purposes only.

File: README.md [hunk 6]
```diff
@@ -30,3 +30,7 @@ See [AUTHORS](AUTHORS.md).
 ## Licensing
 
 hubot-viva-las-vegas is available under the [Apache License, Version 2.0](LICENSE).
+
+## Debug
+
+* `viva reset @username 12.34` comand that overwrites the date of vacation's end where `username` is current user's name and `12.34` is vacation's end date
```
Comment (polina-popova) [inline README.md]: Let's write about it as about test case and take attention, that meeseeks responds to user (as you code) so we need to write his name first.

**Pull request merged**